homeassistant/components/geo_json_events/config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Config flow to configure the GeoJSON events integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from datetime import timedelta
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LOCATION,
+    CONF_LONGITUDE,
+    CONF_RADIUS,
+    CONF_SCAN_INTERVAL,
+    CONF_URL,
+    UnitOfLength,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv, selector
+from homeassistant.util.unit_conversion import DistanceConverter
+
+from .const import (
+    DEFAULT_RADIUS_IN_KM,
+    DEFAULT_RADIUS_IN_M,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): cv.string,
+        vol.Optional(CONF_LOCATION): selector.LocationSelector(
+            selector.LocationSelectorConfig(radius=True, icon="")
+        ),
+        vol.Required(
+            CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+        ): cv.positive_int,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GeoJsonEventsFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a GeoJSON events config flow."""
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        legacy_scan_interval = import_config.get(CONF_SCAN_INTERVAL, None)
+        # Convert scan interval because it now has to be in seconds.
+        if legacy_scan_interval and isinstance(legacy_scan_interval, timedelta):
+            import_config[CONF_SCAN_INTERVAL] = legacy_scan_interval.total_seconds()
```

Comment:
I think it is always a timedelta when imported:
```suggestion
        legacy_scan_interval: timedelta | None = import_config.get(CONF_SCAN_INTERVAL)
        # Convert scan interval because it now has to be in seconds.
        if legacy_scan_interval:
            import_config[CONF_SCAN_INTERVAL] = legacy_scan_interval.total_seconds()
```

Reply:
Yes, that makes sense.